docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,299 @@
+# Process for onboarding new tenants and enrolling them to use Marain services
+
+## Status
+
+Proposed
+
+## Context
+
+We have defined (in ADR 0006) the way in which we intend to implement Tenancy in Marain instances using the `Marain.Tenancy` service. As noted in that ADR, managing the desired model by hand would be excessively error prone and as such, we need to design tooling that will allow us to create and manage new tenants, and to allow them to use the Marain services they are licenced for.
```

Comment:
If would be useful if when you mentioned other ADRs they were hyperlinked for ease of reference.

Reply:
Yes - the problem is that the two related ADRs here are currently in separate PRs! Once one of them is merged back, I'll add the links as part of the other.